### PR TITLE
Options button in header

### PR DESCRIPTION
**Pull request by nienna73**
## What's in this PR:
* added an "Options" button to the header
* added the option to show <e> with or without the macron

### CODE REVIEW

File: validation/jinja2/validation/_base.html [hunk 1]
```diff
@@ -14,12 +15,15 @@
 const csrftoken = document.querySelector('[name=csrfmiddlewaretoken]').value;
 </script>
       <script src="/static/index.js"></script>
+      <script src="/static/menu.js"></script>
```
Comment (eddieantonio) [inline validation/jinja2/validation/_base.html]: I don't understand what's going on with the static stuff, but this is the _cromulent_ way of including static files:

```suggestion
      <script src="{% static 'menu.js' }%"></script>
```

Please see: https://docs.djangoproject.com/en/3.2/howto/static-files/#configuring-static-files

File: validation/static/menu.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+        menu.classList.remove("menu__none");
+        menu.classList.add("menu__block");
+    } else {
+        menu.classList.add("menu__none");
+        menu.classList.remove("menu__block");
+    }
+}
+
+function showMacron(option) {
+    document.cookie = "macron=" + option + "; SameSite=Lax;"
```
Comment (eddieantonio) [inline validation/static/menu.js:15]: Why `SameSite=Lax`?

File: validation/jinja2/validation/_segment_card.html [hunk 3]
```diff
@@ -33,7 +33,13 @@
                 {% if auth %}
                     <button type="button" class="button button--fail button--flag right" data-toggle="modal" data-target="#{{ phrase.id }}-modal" data-cy="flag-button">Something's wrong with this entry</button>
                 {% endif %}
-                <p class="card__title--cree" data-cy="transcription">{{ phrase.transcription }}</p>
+                <p class="card__title--cree" data-cy="transcription">
+                    {% if request.COOKIES['macron'] == 'on' %}
```
Comment (eddieantonio) [inline validation/jinja2/validation/_segment_card.html:37]: If ever you need to support multiple orthographies, I'd take a look at how itwêwina handles this. Ultimately, it's using a cookie to control the orthography, but it has explicit support for multiple orthographies and how to convert to each one.

See: https://github.com/UAlbertaALTLab/cree-intelligent-dictionary/blob/master/CreeDictionary/morphodict/orthography.py

File: validation/jinja2/validation/_menu.html [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+                {% else %}
+                    <li class="menu__options--right" onclick="showMacron('off')"><a href="#" class="menu__options-link">Show e without macron</a></li>
+                {% endif %}
+                <li class="menu__options--right"><a href="/logout" id="logout-link" class="logout-link">Logout</a></li>
+
+            {% else %}
+                <li><a href="/login" id="login-link" class="login-link">Login</a></li>
+            {% endif %}
+        </ul>
+    </div>
```
Comment (eddieantonio) [inline validation/jinja2/validation/_menu.html]: Fun fact: there is a `<menu>` element in the HTML living standard, but it's gone from an old deprecated usage to a new usage (this use case right here). That said, [browers overwhelmingly do not support it](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/menu#browser_compatibility).

For these kind of menus, I have used [`<details>`/`<summary>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) to enable this functionality without JavaScript.

File: validation/jinja2/validation/_segment_card.html [hunk 5]
```diff
@@ -33,7 +33,13 @@
...
+                        {{ phrase.transcription }}
+                    {% else %}
+                        {{ phrase.transcription | replace("ê", "e") }}
```
Comment (eddieantonio) [inline validation/jinja2/validation/_segment_card.html:40]: Pedantic note: the diacritic on the ê [is a circumflex](https://en.wikipedia.org/wiki/Circumflex#Length:~:text=In%20western%20Cree%2C); a macron is [a straight line on top](https://en.wikipedia.org/wiki/Macron_(diacritic)#cite_ref-15:~:text=In%20western%20Cree), like this: ē.

In ALTLab-internal usage, I prefer using the precise names, but I understand that normal folks call both "macrons" — circumflex is such a weird word!

File: validation/static/menu.js (hunk 2 above)
Comment (nienna73) [reply validation/static/menu.js:15]: If I don't provide any sort of CORS information, I get a warning saying that the cookie will expire soon and this gets rid of the warning. I'll do some research to see if there's a better option, though.

File: validation/jinja2/validation/_menu.html (hunk 4 above)
Comment (nienna73) [reply validation/jinja2/validation/_menu.html]: Oh beautiful! I'll see if <details>/<summary> will do the trick!

Comment (nienna73) [reply validation/jinja2/validation/_menu.html]: Okay, I would edit that, but it's just...too good. The HTML got rendered

File: validation/jinja2/validation/_segment_card.html (hunk 5 above)
Comment (nienna73) [reply validation/jinja2/validation/_segment_card.html:40]: This is valid. I was lowkey waiting to see if someone would comment on the wording. I can change it to circumflex and see if we get any user feedback about the wording.

File: validation/static/menu.js (hunk 2 above)
Comment (nienna73) [reply validation/static/menu.js:15]: I think this is actually what we want: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite#lax

Lax is the new default and makes sure the cookie is passed when following a link to the page.

**Review by eddieantonio — APPROVED** [review]
LGTM! Merge it!

**Pull request merged**